Library/Config.ts

```diff
@@ -46,28 +45,26 @@ class Config {
             "*.blob.core.cloudapi.de",
             "*.blob.core.usgovcloudapi.net"];
         
-        this.correlationId = CorrelationIdManager.correlationIdPrefix; // Initialize with a blank correlation ID until we fetch the correct value.
-
         this.setCorrelationId = (correlationId) => this.correlationId = correlationId;
 
-        CorrelationIdManager.queryCorrelationId(
-            this.profileQueryEndpoint,
-            this.instrumentationKey,
-            this.correlationIdRetryIntervalMs,
-            this.setCorrelationId);
+        this.profileQueryEndpoint = process.env[Config.ENV_profileQueryEndpoint] || this.endpointBase;
```

Comment:
I've moved the delay into the `queryCorrelationId` call which is invoked in the setter, so by invoking the setter here we'll queue the fetching of the appId, but allow for it to be cancelled and restarted with a new endpoint if necessary.